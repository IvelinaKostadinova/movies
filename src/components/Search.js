import React from 'react';

const Search = () => {
  return (
    <div className="search-container">
      <input
        id="search-txt"
        type="text"
        placeholder="What do you want to watch?"
      ></input>
      <button id="search-btn" type="submit">
        Search
      </button>
    </div>
  );
};

export default Search;
