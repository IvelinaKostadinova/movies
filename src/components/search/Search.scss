@import '../../styles/variables';

#search {
    padding: 15rem 0 15rem 30rem;

    &__text {
      width: 60rem;
      height: 6rem;
      background-color: $grey;
      text-size-adjust: auto;
      font-size: 2.4rem;
      text-indent: 1.5rem;
      color: $white;
    }

    &__button {
      margin-left: 1rem;
      width: 20rem;
      height: 6rem;
      font-size: 2.5rem;
      color: $white;
      background-color: $pink;
      border-color: $pink;
      border-radius: 0.5rem;
      cursor: pointer;
    }
}