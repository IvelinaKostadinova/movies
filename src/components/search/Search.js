import React, { useState } from 'react';
import PropTypes from 'prop-types';

import './Search.scss';

export const Search = ({ search, onSearch }) => {
  const [text, setText] = useState(search);

  const handleChange = (e) => {
    setText(e.target.value);
  };

  const handleClick = () => {
    onSearch(text);
  };

  return (
    <div id="search">
      <input
        id="search__text"
        type="text"
        placeholder="What do you want to watch?"
        value={text}
        onChange={handleChange}
      ></input>
      <button id="search__button" type="submit" onClick={handleClick}>
        Search
      </button>
    </div>
  );
};

Search.propTypes = {
  search: PropTypes.string.isRequired,
  onSearch: PropTypes.func.isRequired,
};

export default Search;
