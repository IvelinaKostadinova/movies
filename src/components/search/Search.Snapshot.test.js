import React from 'react';
import Search from './Search';
import renderer from 'react-test-renderer';

it('sets Search button option', () => {
  const tree = renderer.create(<Search onSearch={jest.fn()}></Search>);

  expect(tree).toMatchSnapshot();
});
