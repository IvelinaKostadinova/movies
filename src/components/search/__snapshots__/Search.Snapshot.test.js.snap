// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sets MVContextMenu with Edit/Delete option 1`] = `
<div
  id="search"
>
  <input
    id="search__text"
    onChange={[Function]}
    placeholder="What do you want to watch?"
    type="text"
    value=""
  />
  <button
    id="search__button"
    onClick={[Function]}
    type="submit"
  >
    Search
  </button>
</div>
`;

exports[`sets Search button option 1`] = `
<div
  id="search"
>
  <input
    id="search__text"
    onChange={[Function]}
    placeholder="What do you want to watch?"
    type="text"
    value=""
  />
  <button
    id="search__button"
    onClick={[Function]}
    type="submit"
  >
    Search
  </button>
</div>
`;
